### PR TITLE
[QNN EP] Integrate Resize op fixes from QNN 2.14.1

### PR DESCRIPTION
**Pull request by adrianlizarraga**
### Description
QNN SDK version 2.14.1 fixed several issues with the QNN Resize operator. This PR integrates the fixes and simplifies the implementation.

### Motivation and Context
Improve Resize operator and test coverage.

### CODE REVIEW

**Comment by adrianlizarraga** [comment]
@HectorSV @jywu-msft 
This PR is not ready for review, but I wanted to make you aware that QNN 2.14.1 fixed all* of our issues with the Resize operator.

File: onnxruntime/core/providers/qnn/builder/opbuilder/resize_op_builder.cc [hunk 1]
```diff
@@ -122,6 +66,8 @@ class ResizeOpBuilder : public BaseOpBuilder {
   // Arrays of supported QNN modes for QNN's Resize op. The index of each mode is used as the corresponding
   // QNN parameter value. Ex: The "nearest" mode is represented as the value 0 in QNN. Note, that
   // not all modes are supported by every QNN backend.
+  //
+  // TODO: Use enum values from the Qualcomm headers!
```
Comment (adrianlizarraga) [inline onnxruntime/core/providers/qnn/builder/opbuilder/resize_op_builder.cc:70]: Will do this in this PR

File: onnxruntime/test/providers/qnn/resize_test.cc [hunk 2]
```diff
@@ -286,48 +299,103 @@ TEST_F(QnnCPUBackendTests, Resize2xLinearAlignCorners_scales) {
...
+// Expected val: -3.3333334922790527
+// QNN QDQ val: -9.960784912109375 (err 6.6274514198303223)
+// CPU QDQ val: -3.2941176891326904 (err 0.039215803146362305)
+//
+// More debugging info:
+// Input elements f32[1,1,2,2] = -10.0000000 -3.33333349 3.33333302 10.0000000
+// ORT CPU EP (f32 model) outputs: -10.0000000 -10.0000000 -3.33333349 -3.33333349 -3.33333349 -3.33333349 -10.00 ...
+// ORT CPU EP (qdq model) outputs: -9.96078491 -9.96078491 -3.29411769 -3.29411769 -3.29411769 -3.29411769 -9.961 ...
+// ORT QNN EP (qdq model) outputs: -9.96078491 -9.96078491 -9.96078491 -3.37254906 -3.37254906 -3.37254906 -9.961 ...
+TEST_F(QnnHTPBackendTests, DISABLED_ResizeU8_3xNearestAsymmetricRoundPreferFloor) {
```
Comment (adrianlizarraga) [inline onnxruntime/test/providers/qnn/resize_test.cc:387]: We did not previously support this combination of parameters. But I decided to test it now that we have many fixes. Will report to QC.

File: onnxruntime/core/providers/qnn/builder/opbuilder/resize_op_builder.cc [hunk 3]
```diff
@@ -305,92 +201,34 @@
...
-  // QNN's ONNX converter tool translates this configuration (regardless of rounding mode)
-  // to QNN's ResizeNearestNeighbor with {align_corners: 0, half_pixel: 0}.
-  //
-  // NOTE: This is only accurate if the rounding mode is "floor". Need to investigate how to handle
-  // other rounding modes with Qualcomm. Ideally, we would use QNN's Resize operator, but it doesn't support
-  // the "asymmetric" coordinate transformation mode on HTP.
-  if (interp_mode == "nearest" && transformation_mode == "asymmetric") {
+  // Translate Resize with {mode: "nearest", nearest_mode: "floor", coordinate_transformation_mode: XXX} to
+  // QNN's ResizeNearestNeighbor operator on the HTP backend. This combination of parameters is not supported on HTP
+  // via QNN's Resize operator. Note that QNN's ResizeNearestNeighbor operator always uses "floor" rounding.
```
Comment (HectorSVC) [inline onnxruntime/core/providers/qnn/builder/opbuilder/resize_op_builder.cc:215]: Need to report this to QC.

Comment (adrianlizarraga) [reply onnxruntime/core/providers/qnn/builder/opbuilder/resize_op_builder.cc:231]: They already know. It's also in the QNN SDK docs.

Comment (adrianlizarraga) [reply onnxruntime/core/providers/qnn/builder/opbuilder/resize_op_builder.cc:231]: To clarify, the QNN SDK docs state that only the default `nearest_mode` (i.e., round_prefer_floor) is supported for QNN Resize on the HTP backend.

The docs, however, do not state that ResizeNearestNeighbor uses floor rounding. I learned that information from unit tests and from asking QC.

File: onnxruntime/core/providers/qnn/builder/opbuilder/resize_op_builder.cc [hunk 4]
```diff
@@ -143,19 +82,26 @@ const OnnxAttrInfo<std::string> ResizeOpBuilder::onnx_nearest_mode_attr = {"near
...
   }
 
-  return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, "QNN EP: Resize operator does not support ", onnx_mode_label,
-                         " ", std::string(onnx_mode));
+  return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, "QNN EP: Resize operator does not support ", onnx_attr_name,
+                         " ", std::string(onnx_attr_value));
+}
+
+// Returns true if the given ONNX attribute mode value is generally supported on QNN. Note that
+// differnce QNN backends may support a smaller subset of modes.
```
Comment (adrianlizarraga) [inline onnxruntime/core/providers/qnn/builder/opbuilder/resize_op_builder.cc:101]:
```suggestion
// different QNN backends may support a smaller subset of modes.
```

**Comment by adrianlizarraga** [comment]
@HectorSVC  @jywu-msft This is ready for your review when you're available please.

**Review by HectorSVC — APPROVED** [review]
:shipit:

**Pull request merged**